### PR TITLE
make maximum number of retries for username validation configurable

### PR DESCRIPTION
**Pull request by bernt-matthias**
with each retry the username gets username+"-{retry_number}"

rationale: on systems where jobs are submitted as real_user the username
must not be modified in this way. this can now be achieved by
setting max-retries to 0

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

Not sure if there are already tests that I could adapt?

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: lib/galaxy/config/sample/auth_conf.xml.sample [hunk 1]
```diff
@@ -21,6 +21,12 @@
             <!-- Whether Galaxy should automatically register users when they
                  first login. Default is False. -->
 <!--        <auto-register>True</auto-register>
+-->
+
+            <!-- How often Galaxy should retry to validate a user-name: 
+                 If a username is invalid (already taken, to short, to long, non-alphanumeric + '_-')
```
Comment (mvdbeek) [inline lib/galaxy/config/sample/auth_conf.xml.sample]:
```suggestion
                 If a username is invalid (already taken, too short, too long, non-alphanumeric + '_-')
```

**Comment by bernt-matthias** [comment]
I think this also changes the behavior slightly, with two retries it is now: `USER-2` before it was `USER-1-2`, or?

**Review by jmchilton — APPROVED** [review]

**Comment by bernt-matthias** [comment]
I would also be happy with https://github.com/galaxyproject/galaxy/pull/13195 as alternative. So as long an nobody thinks that this may be useful anyway we can close it.

**Review by jdavcs — APPROVED** [review]
API test failure in previous commit was unrelated.

**Comment by mvdbeek** [comment]
haha, just tried to push and got rejected 😆

**Comment by jdavcs** [comment]
> haha, just tried to push and got rejected laughing

Sorry, was trying to be helpful :laughing:

File: lib/galaxy/auth/util.py [hunk 2]
```diff
@@ -78,15 +78,22 @@ def parse_auth_results(trans, auth_results, options):
...
+            test_name = auto_username
         else:
-            raise Conflict("Cannot make unique username")
+            test_name = f"{auto_username}-{try_number}"
+        validate_result = validate_publicname(trans, test_name)
+        if validate_result == "":
+            auto_username = test_name
+            break
+        else:
+            log.error(f"Invalid username '{auto_username}': {validate_result}")
```
Comment (nsoranzo) [inline lib/galaxy/auth/util.py:93]: log.debug may be sufficient?

File: lib/galaxy/config/sample/auth_conf.xml.sample [hunk 3]
```diff
@@ -21,6 +21,12 @@
             <!-- Whether Galaxy should automatically register users when they
                  first login. Default is False. -->
 <!--        <auto-register>True</auto-register>
+-->
+
+            <!-- How often Galaxy should retry to validate a user-name: 
+                 If a username is invalid (already taken, too short, too long, non-alphanumeric + '_-')
+                 Galaxy tries again with username+"-i" for i<= max-retries. -->
+<!--        <max-retries>True</max-retries>
```
Comment (nsoranzo) [inline lib/galaxy/config/sample/auth_conf.xml.sample:29]: s/True/10/

**Review by nsoranzo — COMMENTED** [review]
Sorry for the late review!

File: lib/galaxy/auth/util.py (hunk 2 above)
Comment (jdavcs) [reply lib/galaxy/auth/util.py:93]: I can make the change and push if there are no objections?

File: lib/galaxy/config/sample/auth_conf.xml.sample (hunk 3 above)
Comment (jdavcs) [reply lib/galaxy/config/sample/auth_conf.xml.sample:29]: same as above

File: lib/galaxy/auth/util.py (hunk 2 above)
Comment (mvdbeek) [reply lib/galaxy/auth/util.py:93]: Please make a PR, no need for direct pushes.

**Pull request merged**